src/components/Brushbox-textures.tsx

```diff
@@ -1,14 +1,21 @@
 import { useState } from "react";
 import './Brushbox-textures.css';
 import { Terrain } from "../Model/terrain";
+import { BrushSelection } from "../context";
 
-const BrushBoxTextures = ({ setBrushSelection }) => {
+interface BrushBoxProps {
+    setBrushSelection: (selection: BrushSelection) => void;
+}
+
+
+const BrushBoxTextures = (props: BrushBoxProps) => {
+    const {setBrushSelection} = props
     const [clicked, setClicked] = useState(-1)
     const [key, setKey] = useState(0)
 
     const onClickIcon = (terrain: Terrain) => {
+        setBrushSelection({ terrain });
         setClicked(terrain)
```

Comment:
is clicked something that we still need? I think brush selection probably takes care of everything clicked does. Let's clean this up.